pkg/cli/api_server.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"fmt"
+
+	kms "cloud.google.com/go/kms/apiv1"
+	jvspb "github.com/abcxyz/jvs/apis/v0"
+	"github.com/abcxyz/jvs/internal/version"
+	"github.com/abcxyz/jvs/pkg/config"
+	"github.com/abcxyz/jvs/pkg/justification"
+	"github.com/abcxyz/pkg/cfgloader"
+	"github.com/abcxyz/pkg/cli"
+	"github.com/abcxyz/pkg/logging"
+	"github.com/abcxyz/pkg/serving"
+	"github.com/sethvargo/go-envconfig"
+	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/health"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/reflection"
+)
+
+var _ cli.Command = (*APIServerCommand)(nil)
+
+type APIServerCommand struct {
+	cli.BaseCommand
+
+	// testLookuper overrides the lookuper. It is only used for testing.
+	testLookuper envconfig.Lookuper
+
+	// testBoundAddress is populated by the listener. It is only used for testing
+	// where the server is configured to listen on a raondom port, but tests need
+	// to know the port and address it chose.
+	testBoundAddress string
+}
+
+func (c *APIServerCommand) Desc() string {
+	return `Start an API server`
+}
+
+func (c *APIServerCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+  Start a JVS API server.
+`
+}
+
+func (c *APIServerCommand) Flags() *cli.FlagSet {
+	set := cli.NewFlagSet()
+	return set
+}
+
+func (c *APIServerCommand) Run(ctx context.Context, args []string) error {
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+	args = f.Args()
+	if len(args) > 0 {
+		return fmt.Errorf("unexpected arguments: %q", args)
+	}
+
+	logger := logging.FromContext(ctx)
+	logger.Debugw("server starting",
+		"name", version.Name,
+		"commit", version.Commit,
+		"version", version.Version)
+
+	var cfg config.JustificationConfig
+	if err := cfgloader.Load(ctx, &cfg, cfgloader.WithLookuper(c.testLookuper)); err != nil {
+		return fmt.Errorf("failed to load config: %w", err)
+	}
+
+	grpcServer := grpc.NewServer(grpc.ChainUnaryInterceptor(
+		logging.GRPCUnaryInterceptor(logger, cfg.ProjectID),
+		otelgrpc.UnaryServerInterceptor(),
+	))
+
+	// Create basic health check
+	hs := health.NewServer()
+	hs.SetServingStatus("", healthpb.HealthCheckResponse_SERVING)
+	healthpb.RegisterHealthServer(grpcServer, hs)
```

Comment:
This is a change you'll see throughout. All endpoints expose a super dumb health check, and the tests call the health check and assert a happy response. This is just for CLI testing and is not meant to replace a real integration test.